File: src/lib/api/companyQuickSetup.ts
import { invoke } from "@tauri-apps/api/core";
import type { AppId } from "./types";

export type CompanySetupStatus = "needsConfirmation" | "configured" | "failed";
export type CompanyAppSetupStatus =
  | "pending"
  | "success"
  | "failed"
  | "rolledBack"
  | "skipped";

export interface CompanyKeySetupRequest {
  apiKey: string;
  mode: "default" | "custom";
  apps?: AppId[];
  baseUrl?: string;
  model?: string;
  confirmOverwrite?: boolean;
  resolveUserEnvConflicts?: boolean;
}

export interface ExistingConfig {
  app: string;
  kind: string;
  label: string;
  action: string;
  source?: string;
  resolvable: boolean;
}

export interface AppSetupResult {
  app: string;
  status: CompanyAppSetupStatus;
  message: string;
  rollbackStatus?: string;
}

export interface CompanyKeySetupResult {
  status: CompanySetupStatus;
  existingConfigs: ExistingConfig[];
  appResults: AppSetupResult[];
  warnings: string[];
  backupPath?: string;
  restartRequiredApps: string[];
}

export const companyQuickSetupApi = {
  async quickSetup(
    request: CompanyKeySetupRequest,
  ): Promise<CompanyKeySetupResult> {
    return await invoke("quick_setup_company_key", { request });
  },
};

File: src-tauri/src/commands/company_quick_setup.rs
use tauri::State;

use crate::services::{CompanyKeySetupRequest, CompanyKeySetupResult, CompanyQuickSetupService};
use crate::store::AppState;

#[tauri::command]
pub async fn quick_setup_company_key(
    state: State<'_, AppState>,
    request: CompanyKeySetupRequest,
) -> Result<CompanyKeySetupResult, String> {
    log::info!(
        "[QuickSetup] command received apps={:?} mode={:?} confirm_overwrite={} resolve_env_conflicts={} api_key_len={}",
        request.apps,
        request.mode,
        request.confirm_overwrite,
        request.resolve_user_env_conflicts,
        request.api_key.trim().len()
    );

    match CompanyQuickSetupService::quick_setup_company_key(state.inner(), request).await {
        Ok(result) => {
            log::info!(
                "[QuickSetup] command completed status={:?} app_results={} existing_configs={} warnings={} backup_path={:?}",
                result.status,
                result.app_results.len(),
                result.existing_configs.len(),
                result.warnings.len(),
                result.backup_path
            );
            Ok(result)
        }
        Err(error) => {
            log::error!("[QuickSetup] command failed: {error}");
            Err(error.to_string())
        }
    }
}
